src/libraries/System.Linq/ref/System.Linq.cs

```diff
@@ -56,7 +56,9 @@ public static partial class Enumerable
         public static System.Collections.Generic.IEnumerable<TSource> Except<TSource>(this System.Collections.Generic.IEnumerable<TSource> first, System.Collections.Generic.IEnumerable<TSource> second) { throw null; }
         public static System.Collections.Generic.IEnumerable<TSource> Except<TSource>(this System.Collections.Generic.IEnumerable<TSource> first, System.Collections.Generic.IEnumerable<TSource> second, System.Collections.Generic.IEqualityComparer<TSource>? comparer) { throw null; }
         public static TSource? FirstOrDefault<TSource>(this System.Collections.Generic.IEnumerable<TSource> source) { throw null; }
+        public static TSource? FirstOrDefault<TSource>(this System.Collections.Generic.IEnumerable<TSource> source, TSource? defaultValue) { throw null; }
```

Comment:
I would have expected these `TSource? defaultValue` parameters to instead be `TSource defaultValue`, and then for the return value to also be `TSource` rather than `TSource?`.  If the developer wants to use a nullable, they can define TSource as nullable.

Reply:
These return `TSource?` because these lines are duplicated from the existing `XOrDefault<TSource>` methods. After duplicating the line, I added the `TSource? defaultValue` parameter. The defaultValue parameter is `TSource?` because I presumed that it would be wise to match the signature of the implementation.

Reply:
I'm saying the implementation should use `TSource` rather than `TSource?` as well.  The existing methods use (and need to continue to use) `TSource?` because the default value for a reference type is null, and so it's possible the method could return null.  But with the default value being specified in the new overloads, it's entirely up to the developer whether they want a nullable or non-nullable default value.

Reply:
Ah, I see what you're saying. I'll make some tweaks and push again here shortly.